Sources/Swift/Networking/SentryReachability.swift

```diff
@@ -0,0 +1,232 @@
+import Foundation
+
+#if !os(watchOS) && !((swift(>=5.9) && os(visionOS)) && SENTRY_NO_UIKIT)
+import Network
+
+// MARK: - SentryConectivity
+@objc
+public enum SentryConnectivity: Int {
+    case cellular
+    case wiFi
+    case none
+    
+    func toString() -> String {
+        switch self {
+        case .cellular:
+            return "cellular"
+        case .wiFi:
+            return "wifi"
+        case .none:
+            return "none"
+        }
+    }
+}
+
+@_spi(Private) @objc
+public protocol SentryReachabilityObserver: NSObjectProtocol {
+    @objc func connectivityChanged(_ connected: Bool, typeDescription: String)
+}
+
+// MARK: - SentryReachability
+@_spi(Private) @objc
+public class SentryReachability: NSObject {
+    private var reachabilityObservers = NSHashTable<SentryReachabilityObserver>.weakObjects()
+    private var currentConnectivity: SentryConnectivity = .none
+    private var pathMonitor: Any? // NWPathMonitor for iOS 12+
+    private var reachabilityQueue: DispatchQueue?
+    
+#if DEBUG || SENTRY_TEST || SENTRY_TEST_CI
+    @objc public var skipRegisteringActualCallbacks = false
+    private var ignoreActualCallback = false
+#endif // DEBUG || SENTRY_TEST || SENTRY_TEST_CI
+    
+    @objc(addObserver:)
+    public func add(_ observer: SentryReachabilityObserver) {
+        SentrySDKLog.debug("Adding observer: \(observer)")
+        
+        objc_sync_enter(reachabilityObservers)
+        defer { objc_sync_exit(reachabilityObservers) }
+        
+        SentrySDKLog.debug("Synchronized to add observer: \(observer)")
+        
+        if reachabilityObservers.contains(observer) {
+            SentrySDKLog.debug("Observer already added. Doing nothing.")
+            return
+        }
+        
+        reachabilityObservers.add(observer)
+        
+        if reachabilityObservers.count > 1 {
+            return
+        }
+        
+#if DEBUG || SENTRY_TEST || SENTRY_TEST_CI
+        if skipRegisteringActualCallbacks {
+            SentrySDKLog.debug("Skip registering actual callbacks")
+            return
+        }
+#endif // DEBUG || SENTRY_TEST || SENTRY_TEST_CI
+        
+        let reachabilityQueue = DispatchQueue(label: "io.sentry.cocoa.connectivity")
```

Comment:
`m`: We could consider `SentryDispatchQueueWrapper` even if it is a 1:1 conversion

Reply:
`NWPathMonitor` requires a queue to start